Extension/src/Debugger/configurationProvider.ts

```diff
@@ -34,6 +39,30 @@ abstract class CppConfigurationProvider implements vscode.DebugConfigurationProv
             return undefined;
         }
 
+        // Modify WSL config on 64-bit VSCode for OpenDebugAD7
+        if (os.platform() === 'win32' &&
+            process.arch === 'x64' &&
+            config.pipeTransport &&
+            config.pipeTransport.pipeProgram) {
+            let replacedPipeProgram: string = null;
+            const pipeProgramStr: string = config.pipeTransport.pipeProgram.toLowerCase().trim();
+
+            // OpenDebugAD7 is a 32-bit process. Make sure the WSL pipe transport is using the correct program.
+            replacedPipeProgram = debugUtils.ArchitectureReplacer.checkAndReplaceWSLPipeProgram(pipeProgramStr, debugUtils.ArchType.ia32);
+
+            // If pipeProgram does not get replaced and there is a pipeCwd, concatenate with pipeProgramStr and attempt to replace.
+            if (!replacedPipeProgram && config.pipeTransport.pipeCwd) {
+                const pipeCwdStr: string = config.pipeTransport.pipeCwd.toLowerCase().trim();
+                const newPipeProgramStr: string = path.join(pipeCwdStr, pipeProgramStr);
```

Comment:
What does `path.join` do if you give it to fully-qualified paths?

Reply:
I just checked, and the results are a bit odd. I might suggest first checking if `pipeProgramStr` is relative.

```
> path.join("c:\\", "d:\\jam\\full-path.txt")
'c:\\d:\\jam\\full-path.txt'
```

Reply:
```
path.join(process.env.windir, 'temp')
>'C:\\windows\\temp'
```
I believe path.join is just ```leftParam + path.sep + rightParam```